### PR TITLE
Add support for IAsyncEnumerable #175

### PR DESCRIPTION
**Pull request by cmxl**
Solves #175 

`IAsyncEnumerable` is only available since netstandard2.1.
So I made the `EndpointBase` classes partial, added a new partial class file and removed it from compilation when the targetframework is not netstandard2.1

What I was not sure of, is how to add test for this feature.

### CODE REVIEW

File: src/Ardalis.ApiEndpoints/Ardalis.ApiEndpoints.csproj [hunk 1]
```diff
@@ -25,6 +25,10 @@
     <PackageReference Include="Microsoft.AspNetCore.Mvc.Core" Version="2.1.0" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
   </ItemGroup>
 
+  <ItemGroup Condition=" '$(TargetFramework)' != 'netstandard2.1' ">
+    <Compile Remove="FluentGenerics\EndpointBaseASync.Streams.cs" />
+  </ItemGroup>
+
```
Comment (maxkoshevoi) [inline src/Ardalis.ApiEndpoints/Ardalis.ApiEndpoints.csproj:31]: Don't think this is needed. `#if NETSTANDARD2_1` should be enough

Comment (cmxl) [reply src/Ardalis.ApiEndpoints/Ardalis.ApiEndpoints.csproj]: Thx I just emoved the condition.

**Review by ardalis — APPROVED** [review]

**Comment by cmxl** [comment]
I just added tests and a sample endpoint.
I had to add a direct project reference though, due to automapper failing to load the correct assembly when referencing a multitargeted project. That's also why I needed to add `<SetTargetFramework>TargetFramework=netstandard2.1</SetTargetFramework>` to the project reference in csproj files.
Otherwise the netstandard2.0 version would have been referenced by default.

File: sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj [hunk 2]
```diff
@@ -19,7 +19,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\SampleEndpointApp\SampleEndpointApp.csproj" />
+      <ProjectReference Include="..\..\src\Ardalis.ApiEndpoints\Ardalis.ApiEndpoints.csproj">
+          <SetTargetFramework>TargetFramework=netstandard2.1</SetTargetFramework>
```
Comment (maxkoshevoi) [inline sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj:23]: It doesn't use `netstandard2.1` by default in `net6`? What does it use? Will this be an issue for NuGet consumers?

Comment (maxkoshevoi) [reply sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj:23]: Maybe it makes sense to migrate whole package to `netstandard2.1`. It's only intended to be used in ASP.NET Core anyway

Comment (cmxl) [reply sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj:23]: > Will this be an issue for NuGet consumers?
I remember reading something about it, but did not have time to try.


> Maybe it makes sense to migrate whole package to `netstandard2.1`. It's only intended to be used in ASP.NET Core anyway
I would personally like to upgrade to netstandard2.1.
The compatibility table is here:
https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-1#select-net-standard-version

Comment (maxkoshevoi) [reply sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj:23]: @ardalis How about updating to netstandard2.1?

Comment (cmxl) [reply sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj:23]: @ardalis any thoughts about upgrading to netstandard2.1?

Comment (ardalis) [reply sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj:23]: Yes that's fine with me.

Comment (cmxl) [reply sample/Sample.FunctionalTests/Sample.FunctionalTests.csproj]: #180 has bumped the target framework to netcore3.1, so I removed the netstandard2.1 stuff.
code is so much cleaner now :)

**Pull request merged**